src/helpers/share.helper.ts

```diff
@@ -0,0 +1,62 @@
+import {Observable} from 'rxjs/Observable';
+
+declare let global: any; // To make AoT compiler (ngc) happy
+
+export module Helper {
+
+    export const validateUrl = (url: string, window: Window) => {
+        /** If URL is specified then validate it, otherwise use window URL */
+        if (url) {
+            let r = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/;
+
+            if (r.test(url)) {
+                return url;
+            } else {
+                console.warn('ShareButtons: Invalid URL, switching to window.location.href');
+            }
+        }
+        /** fix window on Universal */
+        return (window) ? encodeURIComponent(window.location.href) :
+            (typeof global !== 'undefined') ? (<any>global).url : '';
+    };
+
+    export const nFormatter = (num, digits) => {
```

Comment:
Yupp, this function was written twice if you have noticed, one in the pipe and one in share-button component, if we are able to get the count in the sharebutton template (without the renderer) then we can move it back to the pipe